proving/postrs/bridge.go

```diff
@@ -0,0 +1,54 @@
+package postrs
+
+// #cgo LDFLAGS: -L/home/bartosz/workspace/post-rs/target/release-clib -lpost
+// #include <stdlib.h>
+// #include "prover.h"
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/spacemeshos/post/config"
+)
+
+type Proof struct {
+	Nonce    uint32
+	Indicies []uint64
+}
+
+func GenerateProof(datadir string, challenge []byte, cfg config.Config) (*Proof, error) {
+	datadirPtr := C.CString(datadir)
+	defer C.free(unsafe.Pointer(datadirPtr))
+
+	challengePtr := C.CBytes(challenge)
+	defer C.free(challengePtr)
+
+	config := C.Config{
+		n:               20,
+		b:               C.uint32_t(cfg.B),
+		k1:              C.uint32_t(cfg.K1),
+		k2:              C.uint32_t(cfg.K2),
+		labels_per_unit: C.uint64_t(cfg.LabelsPerUnit),
+	}
+
+	cProof := C.generate_proof(
+		datadirPtr,
+		(*C.uchar)(challengePtr),
+		C.size_t(len(challenge)),
+		config,
+	)
+
+	if cProof == nil {
+		return nil, fmt.Errorf("got nil")
+	}
+	defer C.free_proof(cProof)
+
+	indicies := unsafe.Slice((*uint64)(unsafe.Pointer(cProof.indicies.ptr)), cProof.indicies.len)
+	indicies = append([]uint64{}, indicies...)
```

Comment:
This seems confusing to me. How about this:

```suggestion
	indicies := make([]uint64, cProof.indicies.len)
	copy(indicies, unsafe.Slice((*uint64)(unsafe.Pointer(cProof.indicies.ptr)), cProof.indicies.len))
```

Reply:
Is the explicit copy even necessary? There is `C.GoBytes` that converts a pointer to a byte array to a `[]bytes` maybe something similar exists for uint64 slices/arrays?

Reply:
I copy it over to Go to be able to free the memory allocated on the Rust side.

Reply:
Copying it is fine and correct; I just advocate to use `copy` over `append` to make it clear that this is a copy 🙂